RecoLocalCalo/HcalRecAlgos/src/ZdcSimpleRecAlgo_Run3.cc

```diff
@@ -0,0 +1,547 @@
+void ZdcSimpleRecAlgo_Run3::initPulseCorr(int toadd, const HcalTimeSlew* hcalTimeSlew_delay) {
+  if (correctForPulse_) {
+    pulseCorr_ = std::make_unique<HcalPulseContainmentCorrection>(
+        toadd, phaseNS_, false, MaximumFractionalError, hcalTimeSlew_delay);
+  }
+}
+
+// helper functions for pedestal subtraction and noise calculations
+namespace zdchelper {
+  inline double subPedestal(const float energy, const float ped, const float width) {
+    if (energy - ped > width)
+      return (energy - ped);
+    else
+      return (0);
+  }
+
+  // assumes that if noise TS is from pileup, noise TS should be
+  // about 40% of previous energy and only 40% of noiseTS will be subtracted from signal
+  inline double calcNoise(const float energy0, const float energy1) {
+    // if(energy0<=0 || energy1<=0)return (energy1);
+    // // check if noise TS is less than half of previous TS
+    // else if( energy0 >= 2*energy1) return( .4*energy1);
+    // else return(.4*energy1);
+    return (.4 * energy1);
+  }
+  inline double calcNoiseRPD(const float energy0, const float energy1) {
+    // if(energy0<=0 || energy1<=0)return (energy1);
+    // // check if noise TS is less than half of previous TS
+    // else if( energy0 >= 2*energy1) return( .4*energy1);
+    // else return(.4*energy1);
+    return (.4 * energy1);
+  }
+
+}  // namespace zdchelper
+
+namespace ZdcSimpleRecAlgoImpl {
+  template <class Digi, class RecHit>
+  inline RecHit reco1(const Digi& digi,
+                      const HcalCoder& coder,
+                      const HcalCalibrations& calibs,
+                      const HcalPedestal& effPeds,
+                      const std::vector<unsigned int>& myNoiseTS,
+                      const std::vector<unsigned int>& mySignalTS) {
+    CaloSamples tool;
+    coder.adc2fC(digi, tool);
+    // Reads noiseTS and signalTS from database
+    int ifirst = mySignalTS[0];
+    double ampl = 0;
+    int maxI = -1;
+    double maxA = -1e10;
+    double ta = 0;
+    double lowGEnergy = 0;
+    double energySOIp1 = 0;
+    double ratioSOIp1 = -1.0;
+    double chargeWeightedTime = 0;
+
+    double Allnoise = 0;
+    int noiseslices = 0;
+    int CurrentTS = 0;
+    double noise = 0;
+    int digi_size = digi.size();
+
+    // determining noise
+    for (unsigned int iv = 0; iv < myNoiseTS.size(); ++iv) {
+      CurrentTS = myNoiseTS[iv];
+      int capid = digi[CurrentTS].capid();
+      // float ped = calibs.pedestal(capid);
+      float ped = effPeds.getValue(capid);
+      float width = effPeds.getWidth(capid);
+      if (CurrentTS >= digi_size)
+        continue;
+      float energy1 = zdchelper::subPedestal(tool[CurrentTS], ped, width);
+      float energy0 = 0;
+      if (CurrentTS > 0)
+        energy0 = zdchelper::subPedestal(tool[CurrentTS - 1], ped, width);
+      Allnoise += zdchelper::calcNoise(energy0, energy1);
+      noiseslices++;
+    }
+    if (noiseslices != 0) {
+      noise = (Allnoise) / double(noiseslices);
+    } else {
```

Comment:
Noise was already initialized to zero so this is not needed.

Reply:
The redundant code has been removed